metrics/prometheus/factory.go

```diff
@@ -57,6 +59,14 @@ func WithBuckets(buckets []float64) Option {
 	}
 }
 
+// WithSeparator returns an option that sets the default separator for the namespace
+// If not used, we fallback to underscore
```

Comment:
- nit: end sentences with dots.
- underscore and colon are the only valid choices for a separator for Prometheus. Given that, another option would be to use an enum (which is better than simply documenting valid values).
```go
type Separator rune
const (
    SeparatorUnderscore rune = '_'
    SeparatorColon           = ':'
)
```

Reply:
I assume the type of `SeparatorUnderscore` is `Separator` instead of `rune`?